### PR TITLE
feat(cstor-operator):introduce cstorpoolcluster resource for pool provisioning.

### PR DESCRIPTION
**Pull request by sonasingh46**
This PR introduces `cstorpoolcluster` resource kind for pool provisioning.
StoragePoolClaim for pool creation will be deprecated in subsequent PR. Although  -- this is not a breaking change.

### CODE REVIEW

File: pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go [hunk 1]
```diff
@@ -24,72 +24,72 @@ import (
...
-type StoragePoolClaimStatus struct {
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
 	Phase string `json:"phase"`
 }
 
-// CStorPoolAttr is to describe zpool related attributes.
-type CStorPoolAttr struct {
+// CStorPoolClusterSpecAttr is to describe zpool related attributes.
+type CStorPoolClusterSpecAttr struct {
```
Comment (AmitKumarDas) [inline pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go]: Can we use string constant as datatype for PoolType?

File: pkg/cstorpoolcluster/v1alpha1/cstorpoolcluster.go [hunk 2]
```diff
@@ -18,19 +18,18 @@ package v1beta1
 
 import (
 	apisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
-	apisv1beta1 "github.com/openebs/maya/pkg/apis/openebs.io/v1beta1"
 )
 
-// SPC encapsulates StoragePoolClaim api object.
+// SPC encapsulates CStorPoolCluster api object.
```
Comment (AmitKumarDas) [inline pkg/cstorpoolcluster/v1alpha1/cstorpoolcluster.go]: Can we rename this struct to CSPC & similarly other names wherever applicable?

Comment (sonasingh46) [reply pkg/cstorpoolcluster/v1alpha1/cstorpoolcluster.go]: Yes. Will fix it.

File: pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go (hunk 1 above)
Comment (sonasingh46) [reply pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go]: Will fix it.

Comment (sonasingh46) [reply pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go]: fixed.

**Pull request merged**